src/pybind/mgr/mgr_module.py

```diff
@@ -920,6 +940,184 @@ def cluster_log(self, channel: str, priority: ClusterLogPrio, message: str) -> N
     def version(self) -> str:
         return self._version
 
+    def pool_exists(self, name: str) -> bool:
+        pools = [p['pool_name'] for p in self.get('osd_map')['pools']]
+        return name in pools
+
+    def have_enough_osds(self) -> bool:
+        # wait until we have enough OSDs to allow the pool to be healthy
+        up = 0
+        for osd in self.get("osd_map")["osds"]:
+            if osd["up"]:
```

Comment:
you probably want up + in, to ensure they're available for data placement as well as running

Reply:
done